modules/vault-iam/main.tf

```diff
@@ -0,0 +1,44 @@
+resource "aws_iam_user" "vault_user" {
+  name = "vault_user"
+  tags = {
+    "user" = "vault"
+  }
+}
+
+resource "aws_iam_access_key" "vaultkey" {
+  user = "${aws_iam_user.vault_user.name}"
+}
+
+resource "aws_iam_user_policy" "vault_user_policy" {
```

Comment:
We could reuse the general idea here, so I would recommend we consider renaming the module be a generic module, which we can then use in your vault example with the `vault` namespace. Alternatively, if this module needs to have a custom list of actions, we could create a generic module that provides "admin on IAM within this namespace", where the namespace is provided as a variable, and separate from this `vault-iam` module.

Reply:
I have extracted a generic module named `iam-user-policy` out of this. I have moved the `vault-iam` module under `examples` where it uses this module to create an user and an associated policy.